docs/rfcs/0003_templates.md

```diff
@@ -0,0 +1,353 @@
+-   If `fieldsSchema` is supplied then use that to populate
+    `TemplateVersion.fields_schema`.
+-   Otherwise, if `fieldsSchemaFile` it is supplied then parse it into
+    JSON and use that to populate `TemplateVersion.fields_schema`.
+-   At least one of these must be supplied
+
+### CreateTemplateMutation
+
+Creates a Template and initial TemplateVersion.
+
+```graphql
+type CreateTemplateInput {
+    name: String!
+    dataType: String!
+    description: String!
+    fieldsSchema: JSONString
+    fieldsSchemaFile: Upload
+}
+```
+
+### CreateTemplateVersionMutation
+
+Creates a TemplateVersion.
+
+```graphql
+type CreateTemplateVersionInput {
+    submissionTemplate: ID!
+    name: String!
+    description: String!
+    fieldsSchema: JSONString
+    fieldsSchemaFile: Upload
+}
+```
+
+### CopyTemplateMutation
+
+Creates a new Template and a TemplateVersion by duplicating the
+referenced template version and it's submission template. Overwrites both
+the name and description with the supplied values.
+
+```graphql
+type CopyTemplateInput {
+    templateVersion: ID!
+    name: String!
+    description: String!
+}
+```
+
+### Permissions
+
+-   Any organization or study administrator may create a template
+
+## Reviewing Templates
+
+At some point in the future we may want to consider implementing a collaborative
+review process to manage the development of templates. This is TBD for now.
+
+## Releasing Templates
+
+In order for a template to be used in studies, it must be released/approved
+for use via a TemplatePackage and then assigned to a study. The workflow for
+this would be:
+
+1. Create a template package
+    - Add name, description
+    - Add template versions
+    - Assign version number
+2. Assign the template package to a study
+
+This allows us to track the provenance of submission requirements
+for each study.
+
+### CreateTemplatePackageMutation
+
+Creates a TemplatePackage and optionally assigns it to a study
+
+```graphql
+type CreateTemplatePackageInput {
+    name: String!
+    description: String!
+    templateVersions: [ID]!
+    version: String
+    study: ID
+}
+```
+
+### UpdateTemplatePackageMutation
+
+Updates a TemplatePackage and optionally assigns it to a study.
+
+```graphql
+type UpdateTemplatePackageInput {
+    id: ID!
+    name: String
+    description: String
+    templateVersions: [ID]
+    version: String
+    study: ID
+}
+```
+
+### DeleteTemplatePackageMutation
+
+Deletes a TemplatePackage. Only TemplatePackages that have not been assigned
+to any studies may be deleted.
+
+```graphql
+type DeleteTemplatePackageInput {
+    id: ID!
+}
+```
+
+### Permissions
+
+-   Any organization or study administrator may create a template
+    package.
+-   Any organization or study administrator may assign a
+    template package to a study that they have access to.
+-   Any organization or study administrator may update a template
+    package.
+-   Any organization or study administrator may delete a template
```

Comment:
Will they be able to create template packages they didn't create?

Reply:
I think you're looking at an old version of the RFC? I got rid of the `0003_templates.md` doc